cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/sqlite/android/AndroidSQLite.java

```diff
@@ -33,7 +33,9 @@
 
 
     public static AndroidSQLite createAndroidSQLite(String path) {
-        SQLiteDatabase db = SQLiteDatabase.openDatabase(path, null, SQLiteDatabase.CREATE_IF_NECESSARY);
+        SQLiteDatabase db = (path != null)
+                ? SQLiteDatabase.openDatabase(path, null, SQLiteDatabase.CREATE_IF_NECESSARY)
```

Comment:
I tend to prefer the ternary operator for cases like this where one of two values are being assigned, but I don't mind much the outcome is the same.
Changed to if/else in 07aedeee952055d401a4bea457ed65f4371af37c along with the complementary assignment in `SQLiteWrapper`.